backend/app/main.py

```
@@ -16,13 +16,18 @@
+from cachetools import TTLCache
 
 from app.redaction import EnhancedPIIRedactor
 from app import __version__
 
 # Track process start time for uptime
 PROCESS_START_TIME = time.time()
 
+# Thread-safe TTL cache for redaction results
+# Cache size: 100 entries, TTL: 1 hour (3600 seconds)
+REDACTION_CACHE = TTLCache(maxsize=100, ttl=3600)
+REDACTION_CACHE_LOCK = threading.RLock()
```

Ruff (0.12.2)

Undefined name `threading`

(F821)